tests/integration/tier0/backup-release/test_backup_release.py

```diff
@@ -1,37 +1,193 @@
 import os
-import platform
+import re
+
+from collections import namedtuple
+
+import pytest
 
 from envparse import env
 
 
-system = platform.platform()
+# TODO move to conftest
+class GetSystemVersion:
+    """
+    Helper class.
+    Assign a namedtuple with major and minor elements, both of an int type.
+    Assign a distribution (e.g. centos, oracle, rocky, alma0
+    Assign a system release.
+
+    Examples:
+    Oracle Linux Server release 6.10
+    Oracle Linux Server release 7.8
+    CentOS release 6.10 (Final)
+    CentOS Linux release 7.6.1810 (Core)
+    CentOS Linux release 8.1.1911 (Core)
+    """
+
+    with open("/etc/system-release", "r") as file:
+        system_release_content = file.read()
+        match_version = re.search(r".+?(\d+)\.(\d+)\D?", system_release_content)
+        if not match_version:
+            print("not match")
+        version = namedtuple("Version", ["major", "minor"])(int(match_version.group(1)), int(match_version.group(2)))
+        distribution = system_release_content.split()[0].lower()
+        system_release = "{}-{}.{}".format(distribution, version.major, version.minor)
 
 
-def test_backup_os_release_no_envar(shell, convert2rhel):
+class AssignRepositoryVariables:
```

Comment:
The name of the class doesnt say much about what it actually does. For example I dont see the connection between the name and the `backup_dir` variable.

Is this class needed at all?

Reply:
Removed the class and made it a function